server/src/main/java/org/elasticsearch/index/query/SpanOrQueryBuilder.java

```diff
@@ -58,7 +59,7 @@ public SpanOrQueryBuilder(SpanQueryBuilder initialClause) {
     public SpanOrQueryBuilder(StreamInput in) throws IOException {
         super(in);
         for (QueryBuilder clause: readQueries(in)) {
-            clauses.add((SpanQueryBuilder) clause);
+            addClause((SpanQueryBuilder) clause);
```

Comment:
I think this is not necessary.  This comes from another node that presumably already failed a span query if it contains boost.
The only place where we should do the check is inside `fromXContent` method for all span query builders : `SpanOrQueryBuilder`, `SpanTermQueryBuilder` etc